### PR TITLE
ZIOS-11281: Delivery receipt optimization - sync engine support

### PR DESCRIPTION
**Pull request by nicorsm**
## What's new in this PR?

Added support for DM changes about delivery receipt optimization:
- Added call to `ZMConversation.confirmDeliveredMessages` during the event processing
- Changed tests in order to support the changes (since we're appending the delivery receipt in case of new messages there is a new hidden message appended to the conversation. Open to suggestions)

### Needs release with

- [x] https://github.com/wireapp/wire-ios-data-model/pull/699

### CODE REVIEW

File: Source/Synchronization/ZMSyncStrategy+EventProcessing.swift [hunk 1]
```diff
@@ -49,6 +49,12 @@ extension ZMSyncStrategy: ZMUpdateEventConsumer {
                 self.eventProcessingTracker?.registerEventProcessed()
             }
             localNotificationDispatcher?.processEvents(decryptedUpdateEvents, liveEvents: true, prefetchResult: nil)
+            
+            if let messages = fetchRequest.noncesToFetch.allObjects as? [UUID],
```
Comment (typfel) [inline Source/Synchronization/ZMSyncStrategy+EventProcessing.swift]: Isn't this now bypassing `BackgroundAPNSConfirmationStatus`?

Comment (nicorsm) [reply Source/Synchronization/ZMSyncStrategy+EventProcessing.swift]: Mmh seems so. We should check for `BackgroundAPNSConfirmationStatus.sendDeliveryReceipts`, right?

Comment (typfel) [reply Source/Synchronization/ZMSyncStrategy+EventProcessing.swift]: We can also modify `BackgroundAPNSConfirmationStatus` suit or needs.

File: Tests/Source/Synchronization/Transcoders/ZMConversationTranscoderTests.m [hunk 2]
```diff
@@ -2091,6 +2091,7 @@ - (void)testThatItMarksTheConversationAsNeedingToBeUpdatedWhenTheSelfUserIsAdded
         }
         
         // then
+        XCTAssertEqualObjects(conversation.keysThatHaveLocalModifications, [NSSet setWithObject:ZMConversationArchivedChangedTimeStampKey]);
```
Comment (typfel) [inline Tests/Source/Synchronization/Transcoders/ZMConversationTranscoderTests.m]: We shouldn't add this back it's not relevant for the test.

File: Tests/Source/Integration/APNSTests.m [hunk 3]
```diff
@@ -238,7 +238,7 @@ - (void)testThatItSendsAConfirmationMessageWhenReceivingATextMessage
             NSString *clientsPath = [NSString stringWithFormat:@"/users/prekeys"];
             if ([request.path isEqualToString:clientsPath]) {
                 XCTAssertTrue(request.shouldUseVoipSession);
-                XCTAssertEqual(requestCount, 1u);
+                XCTAssertEqual(requestCount, 2u);
```
Comment (typfel) [inline Tests/Source/Integration/APNSTests.m]: Any idea why this changed?

**Comment by typfel** [comment]
Do we have an integration test testing that we send delivery receipts?

**Comment by nicorsm** [comment]
@zenkins release minor

**Pull request merged**